source/Octopus.Tentacle.Tests.Integration/Util/TcpTentacleHelpers/ListeningResponseMessageTcpKiller.cs

```diff
@@ -40,18 +40,26 @@ public IDataTransferObserver DataTransferObserver()
         {
             return new DataTransferObserverBuilder().WithWritingDataObserver((tcpPump, dataFromTentacle) =>
             {
-                var size = dataFromTentacle.Length;
-                // It seems messages around 45 and below are control messages
+                // It seems messages around 45 and below are control messages - except for
+                // Windows Server 2012, where the max size seems to be ~85.
                 // So anything bigger must be the interesting one.
-                if (pauseConnection && size > 45)
+                //
+                // We increased this value from 45 -> 85 to account for Windows Server 2012,
+                // and all the integration test builds seem happy with it.
+                // May need to be adjusted in the future to account for different OSs,
+                // runtimes, lunar cycles, and/or any other random heuristic.
+                const int assumedControlMessageMaxSize = 85;
+                
```

Comment:
Mmmmmmm... magic numbers!

We had some tests reliably failing on 2012/2012 R2 builds, but only when:
* Tentacle was in listening mode
* Test tries to kill the connection

Turns out almost everything was working as expected, _except_ these tests were killing the connection at _very slightly the wrong time_ due to the control messages being a different size on 2012. Killing the connection at this slightly-wrong point somehow resulted in the connection recovering automatically and then completing the RPC call successfully. Fantastic, robust behaviour! Except, of course, we are trying to simulate a flaky network and ensure RPC retries are doing what they are supposed to do, so the tests fail.

We discussed increasing this magic number for *only* Windows Server 2012/2012 R2 integration test builds, but all the other integration test runs are happy with this larger number, so it seems acceptable to change it across the board.